<launch>
  <arg name="map_file" />
  <arg name="map_name" />
  <arg name="map_frame_id" />

  <!-- Remap port/host as required -->
  <param name="warehouse_host" value="localhost"/>
  <param name="warehouse_port" value="27017"/>

  <node name="mongo" type="mongo_wrapper_ros.py" pkg="warehouse_ros">
    <param name="overwrite" value="true"/>
    <param name="database_location" value="warehouse_data"/>

  </node>

  <node pkg="map_store" type="map_saver" name="map_saver" />
  <node pkg="map_store" type="map_manager" name="map_manager" output="screen">
     <!-- unused topics -->
     <remap from="dynamic_map" to="map_store_dynamic_map"/>
     <remap from="map" to="map_store_map"/>
  </node> 

  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
     <!-- Remaps to load the map file -->
     <remap from="static_map" to="dynamic_map" />
     <remap from="map" to="map_store/map" />
     <param name="frame_id" value="$(arg map_frame_id)" />
  </node>

  <node pkg="map_store" type="add_map.py" name="add_map" args="'$(arg map_name)'" required="true" />


</launch>